qcodes/instrument_drivers/QDev/QDac.py

```diff
@@ -0,0 +1,484 @@
+                'read_state valid params are {}'.format(valid_params))
+
+        self._get_status(readcurrents=False)
+
+        parameter = 'ch{:02}_{}'.format(chan, param)
+        value = self.parameters[parameter].get_latest()
+
+        returnmap = {'vrange': {1: 1, 10: 0},
+                     'irange': {0: '1 muA', 1: '100 muA'}}
+
+        if 'range' in param:
+            value = returnmap[param][value]
+
+        return value
+
+    def _get_status(self, readcurrents=False):
+        r'''
+        Function to query the instrument and get the status of all channels.
+        Takes a while to finish.
+
+        The `status` call generates 51 lines of output. Send the command and
+        read the first one, which is the software version line
+        the full output looks like:
+        Software Version: 0.160218\r\n
+        Channel\tOut V\t\tVoltage range\tCurrent range\n
+        \n
+        8\t  0.000000\t\tX 1\t\tpA\n
+        7\t  0.000000\t\tX 1\t\tpA\n
+        ... (all 48 channels like this in a somewhat peculiar order)
+        (no termination afterward besides the \n ending the last channel)
+        returns a list of dicts [{v, vrange, irange}]
+        NOTE - channels are 1-based, but the return is a list, so of course
+        0-based, ie chan1 is out[0]
+        '''
+
+        # Status call
+
+        version_line = self.ask('status')
+
+        if version_line.startswith('Software Version: '):
+            self.version = version_line.strip().split(': ')[1]
+        else:
+            self._wait_and_clear()
+            raise ValueError('unrecognized version line: ' + version_line)
+
+        header_line = self.read()
+        headers = header_line.lower().strip('\r\n').split('\t')
+        expected_headers = ['channel', 'out v', '', 'voltage range',
+                            'current range']
+        if headers != expected_headers:
+            raise ValueError('unrecognized header line: ' + header_line)
+
+        chans = [{} for i in self.chan_range]
+        chans_left = set(self.chan_range)
+        while chans_left:
+            line = self.read().strip()
+            if not line:
+                continue
+            chanstr, v, _, vrange, _, irange = line.split('\t')
+            chan = int(chanstr)
+            chanstr = '{:02}'.format(chan)
+
+            irange_trans = {'hi cur': 1, 'lo cur': 0}
+
+            vals_dict = {
+                'vrange': ('ch{}_vrange',
+                           self.voltage_range_status[vrange.strip()]),
+                'irange': ('ch{}_irange', irange_trans[irange]),
+                'v': ('ch{}_v', float(v)),
+            }
+
+            chans[chan - 1] = vals_dict
+            for param in vals_dict:
+                parameter = vals_dict[param][0].format(chanstr)
+                value = vals_dict[param][1]
+                if param == 'vrange':
+                    attenuation = 0.1*value
+                if param == 'v':
+                    value *= attenuation
```

Comment:
It's not clear to me that `attenuation` is defined if param is `v` does `vrange` always come before v in vals_dict